gateway/1.4/modules/ROOT/pages/_partials/task-node-affinity.adoc

```diff
@@ -0,0 +1,225 @@
+
+Configure node affinity rules within a Pod spec. A Pod spec is YAML file in which you can specify rules for required, preferred, or both types of matching:
+
+* `requiredDuringSchedulingIgnoredDuringExecution`: The node must match all node affinity rules configured for a Pod when the Kubernetes scheduler applies this type of matching. Otherwise, the deployment fails. 
+* `preferredDuringSchedulingIgnoredDuringExecution`: Based on weighted preferences set for a Pod, the Kubernetes scheduler selects the best matching node. The scheduler must select a node, so if no node matches any of the preferences, the scheduler selects a node that does not match. 
+
+For guidance, see the following configuration options:
+
+* <<strict_match_only>>
+* <<soft_match_only>>
+* <<all_match_types>>
+
+[[strict_match_only]]
+=== Configure a Required Node Matching Rule for a Pod 
+
+In required node matching, the node must meet all node affinity rules configured for a Pod. Otherwise, the deployment fails.
+//end::temp-tag1[]
+
+//TAG for common node affinity settings
+//tag::config-flex-affinity-condition[]
+Configure a node affinity rule in the Pod spec file. Each condition to a rule requires a node label as a `key`, an operator, and one or more node label values to accept for a node with that key. For operators, you can use `In`, `NotIn`, `Exists`,  `DoesNotExist`, `Gt`, and `Lt`.
+//end::config-flex-affinity-condition[]
+
+[source,yaml,subs=attributes+]
+----
+affinity:
+ nodeAffinity:
+   requiredDuringSchedulingIgnoredDuringExecution:
+     nodeSelectorTerms:
+     - matchExpressions:
+       - key: <node-label-key>
+         operator: <operator>
+         values:
+         - <node-label-value>
+       # - Add any other node label values. 
+    # - Add any other required conditions.
+----
+
+You can add as many values and conditions as the node requires. 
+
+//TAG for upgrading pod YAML spec
+//tag::upgrade-flex-pod-spec-yaml[]
+To incorporate the node affinity settings from your Pod spec into the Helm chart, you need to use a Helm command. https://helm.sh/docs/intro/install/[Helm^], a tool used to install Flex Gateway, monitoring tools, and applications. A minimum Helm version of 3.0.0 is required. 
+
+The Helm command to use depends on whether Flex Gateway is installed:
+
+* If you are installing Flex Gateway for the first time, use a command like this one for the installation:
++
+[source,kubernetes,subs=attributes+]
+----
+helm -n gateway upgrade -i --create-namespace --wait ingress flex-gateway/flex-gateway -f <path-to-yaml-file> --set-file registration.content=<path-to-registration>
+----
++
+Notice that the command passes the YAML file for the Pod spec.
+
+* If Flex Gateway is installed already, use a command like this one to your running instance:
++
+[source,kubernetes,subs=attributes+]
+----
+helm -n gateway upgrade -i --create-namespace --wait ingress flex-gateway/flex-gateway --reuse-values -f <path-to-yaml-file>
```

Comment:
same Q re breaking up the command in to multiple lines

Reply:
same answer :P

Reply:
Here I would add that the difference between both commands is that this one reuses all the existing configuration, and the one above resets all values

Reply:
```suggestion
helm -n gateway upgrade -i --create-namespace \
--wait ingress flex-gateway/flex-gateway \
--reuse-values -f <path-to-yaml-file>
```